offload/liboffload/src/OffloadImpl.cpp

```diff
@@ -247,11 +244,50 @@ Error olGetDeviceInfoImplDetail(ol_device_handle_t Device,
                                DevInfo.getQueue().end(), InfoKeyMatches);
 
       if (Item != std::end(DevInfo.getQueue())) {
-        return Item->Value;
+        return Item;
       }
     }
 
-    return std::string("");
+    return std::nullopt;
+  };
+  auto GetInfoString = [&](std::vector<std::string> Names) {
```

Comment:
We should pass `Names` by reference here and in `GetInfoXyz` (I think this is my mistake from the original implementation)

Is there a reason we can't use `StringRef` and `SmallVector` here as well?

Reply:
It looks like trying to pass it by reference results in us being unable to use `{"Device Name"}` in the literal list (because it isn't an lvalue reference).

Reply:
Ah, that makes sense, never mind then